src/app/auth/login/login.component.ts

```diff
@@ -0,0 +1,36 @@
+import { Component } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import { Router } from '@angular/router';
+import { map, Subscription } from 'rxjs';
+import { AuthService } from '../auth.service';
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.scss'],
+})
+export class LoginComponent {
+  unsubscribe$!: Subscription;
```

Comment:
```suggestion
  unsubscribe$: Subscription;
```

and need `this.unsubscribe$.unsubscribe()`